docs(hiring): add details for systems design interview (#44)
loadsmart/culture: hiring/interviewing.md

## Systems Design interview
After finishing the [coding exercise](#coding-exercise), the review will happen either automatically (if done via HackerRank) or manually (if done via a GitHub repo). In case of a satisfactory solution, the next step is the Systems Design Interview.

> If in the previous interview you had the chance to show your coding and logical-thinking abilities, in this interview, you will be demonstrating a bit of your thought process when designing new systems given a problem that is described to you. This is supposed to be an interactive conversation. Feel free to ask any questions during the interview if you need clarification on any topic. 

Reviewer: ```suggestion
In the previous interview you had the chance to show your coding and logical-thinking abilities. In this interview, you will be demonstrating a bit of your thought process when designing new systems given a problem that is described to you. This is supposed to be an interactive conversation. Feel free to ask any questions during the interview if you need clarification on any topic. 
```

We will provide a list of high-level requirements, and we expect that you:

> - understand and discuss the given requirements, making sure there are no doubts

Reviewer: ```suggestion
- Understand and discuss the given requirements, making sure clarify anything that isn't clear to you
```

> - propose a high-level design and scope for your solution, explaining each of your decisions

Reviewer: ```suggestion
- Propose a high-level design and scope for your solution, explaining each of your decisions
```

> - can deep-dive into the implementation details of each of the components you propose

Reviewer: ```suggestion
- Delve into the implementation details of each of the components you propose
```

> - identify the bottlenecks of your solution and point areas of improvement

Reviewer: ```suggestion
- Identify potential bottlenecks with your solution and recommend ways to improve the solution
```
